pkg/policy/parser.go

```diff
@@ -96,9 +96,26 @@ func decodePolicyContent(labels []string, content *hcl.BodyContent, ctx *hcl.Eva
 		}
 		innerContent, contentDiags := body.Content(policySchema)
 		diags = append(diags, contentDiags...)
+		if innerContent == nil {
+			return nil, diags
+		}
 		inner, innerDiags := decodePolicyContent([]string{""}, innerContent, ctx, r)
 		diags = append(diags, innerDiags...)
-		policy.Policies = append(policy.Policies, inner.Policies...)
+		if inner == nil {
```

Comment:
why this logic and not `diags.HasError`? `decodePolicyContent` should make sure to not return a nil `inner` unless diags is set maybe, if it doesn't already?